node/network/gossip-support/src/lib.rs

```diff
@@ -0,0 +1,167 @@
+// Copyright 2021 Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+//! This subsystem is responsible for keeping track of session changes
+//! and issuing a connection request to the validators relevant to
+//! the gossiping subsystems on every new session.
+
+use futures::FutureExt as _;
+use polkadot_node_subsystem::{
+	messages::{
+		GossipSupportMessage,
+	},
+	ActiveLeavesUpdate, FromOverseer, OverseerSignal,
+	Subsystem, SpawnedSubsystem, SubsystemContext,
+};
+use polkadot_node_subsystem_util::{
+	validator_discovery::{ConnectionRequest, self},
+	self as util,
+};
+use polkadot_primitives::v1::{
+	Hash, ValidatorId, SessionIndex,
+};
+use polkadot_node_network_protocol::peer_set::PeerSet;
+
+const LOG_TARGET: &str = "gossip_support";
+
+/// The Connection Request Issuer subsystem.
+pub struct GossipSupport {}
```

Comment:
```suggestion
pub struct GossipSupport;
```

Reply:
I'll add some basic metrics as a followup to address p.2.